src/source/PeerConnection/PeerConnection.c

```diff
@@ -179,10 +177,36 @@ STATUS sendPacketToRtpReceiver(PKvsPeerConnection pKvsPeerConnection, PBYTE pBuf
         pTransceiver = (PKvsRtpTransceiver) item;
 
         if (pTransceiver->jitterBufferSsrc == ssrc) {
+            packetsReceived++;
+            if (STATUS_FAILED(retStatus = decryptSrtpPacket(pKvsPeerConnection->pSrtpSession, pBuffer, (PINT32) &bufferLen))) {
+                DLOGW("decryptSrtpPacket failed with 0x%08x", retStatus);
+                packetsFailedDecryption++;
+                CHK(FALSE, STATUS_SUCCESS);
+            }
+            now = GETTIME();
             CHK(NULL != (pPayload = (PBYTE) MEMALLOC(bufferLen)), STATUS_NOT_ENOUGH_MEMORY);
             MEMCPY(pPayload, pBuffer, bufferLen);
             CHK_STATUS(createRtpPacketFromBytes(pPayload, bufferLen, &pRtpPacket));
-            CHK_STATUS(jitterBufferPush(pTransceiver->pJitterBuffer, pRtpPacket));
+            pRtpPacket->receivedTime = now;
+
+            // https://tools.ietf.org/html/rfc3550#section-6.4.1
+            // https://tools.ietf.org/html/rfc3550#appendix-A.8
+            // interarrival jitter
+            // arrival, the current time in the same units.
+            // r_ts, the timestamp from   the incoming packet
+            arrival = KVS_CONVERT_TIMESCALE(GETTIME(), HUNDREDS_OF_NANOS_IN_A_SECOND, pTransceiver->pJitterBuffer->clockRate);
+            r_ts = pRtpPacket->header.timestamp;
+            transit = arrival - r_ts;
+            d = transit - pTransceiver->pJitterBuffer->transit;
```

Comment:
Any possibility `d` evaluates to a negative number?

Reply:
let me doublecheck

Reply:
changed to signed